meteorite_fall.js

```diff
@@ -49,6 +54,25 @@ function initializeMeteorites() {
     }
 }
 
+function checkGameOver(meteoriteX, meteoriteY, playerX, playerY, meteoriteWidth, meteoriteHeight, playerWidth) {
+    if (meteoriteY + meteoriteHeight == playerY) {
+        if (meteoriteX <= playerX + playerWidth && meteoriteX + meteoriteWidth >= playerX) {
+            isGameOver = true;
+            context.clearRect(0,0, canvasWidth, canvasHeight);
+            resetMeteorites();
+            alert("Game Over!");
+        }
+    }
+}
+
+function resetMeteorites() {
+    for (arrayIndex = 0; arrayIndex < meteorites.length; arrayIndex++) {
```

Comment:
I would just use 'i' as 'arrayIndex' basically is just indicating we are using it as an iterator. and i is the standard. I would things other than i when the actual name gives more info. For example if you had a slow, and fast pointer, etc...

Reply:
Updated.